src/matcha_ml/services/analytics_service.py

```diff
@@ -30,10 +30,43 @@ class AnalyticsEvent(str, Enum):
     GET = "get"
 
 
-def execute_analytics_event(
+def _get_state_uuid() -> MatchaResourceProperty:
+    """A function for retrieving the Matcha State UUID.
+
+    Returns:
+        matcha_state_uuid (MatchaResourceProperty): The Matcha State UUID if present.
+
+    Raises:
+    MatchaError where the Matcha state UUID fails validation.
+    """
+    try:
+        matcha_state_service = MatchaStateService()
+    except MatchaError:
+        matcha_state_service = None
+
+    if matcha_state_service and matcha_state_service.state_exists():
```

Comment:
What happens in the case where the `matcha_state_service.state_exists()` returns false? This function will likely fail as the `matcha_state_uuid` hasn't be created (reference before setting). If this if statement evaluates to false, then I'd raise an exception. 

Your tests should catch this particular case as you want to test for conditionals in your code (this is where functional programming comes in handy, with smaller building blocks)

Reply:
TBF the `.state_exists()` test is now redundant (if we're ignoring pre-analytics installs as discussed).

If state doesn't exist, that's handled in the instantiation of the MatchaStateService class.

I did have a test for the `None` return case but removed it when we discussed removing the optionality for the return yesterday. 

If we're expecting this utility function to always return a MatchaStateResource, then a lot of the conditional logic can be stripped out. I'll do this now.